.github/container/Dockerfile.pax.arm64

```diff
@@ -1,65 +1,160 @@
-# We need to build some packages from source, bring some dependencies.
-RUN apt-get update && \
-  apt-get update && \
-  apt-get install -y \
-    bat \
-    curl \
-    git \
-    gnupg \
-    rsync \
-    liblzma-dev \
-    && \
-  apt-get autoremove -y && apt-get clean && rm -rf /var/lib/apt/lists
+ARG REPO_PAXML=https://github.com/google/paxml.git
+ARG REPO_PRAXIS=https://github.com/google/praxis.git
+ARG REF_PAXML=main
+ARG REF_PRAXIS=main
+ARG SRC_PATH_PAXML=/opt/paxml
+ARG SRC_PATH_PRAXIS=/opt/praxis
 
+###############################################################################
+## build tensorflow-text and lingvo, which do not have working arm64 pip wheels
+###############################################################################
 
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE} as wheel-builder
+
+# tensorflow-text and lingvo build needs bazel
 RUN wget https://github.com/bazelbuild/bazelisk/releases/download/v1.17.0/bazelisk-linux-arm64 -O /usr/bin/bazel && \
     chmod a+x /usr/bin/bazel
 
+#------------------------------------------------------------------------------
+# build tensorflow-text 2.13.0 from source
+#------------------------------------------------------------------------------
+
+FROM wheel-builder as tftext-builder
+
+RUN <<"EOT" bash -exu
+set -o pipefail
+pip install tensorflow_datasets==4.9.2 auditwheel tensorflow==2.13.0
+git clone http://github.com/tensorflow/text.git /opt/tensorflow-text
+cd /opt/tensorflow-text
+git checkout v2.13.0
+./oss_scripts/run_build.sh
+EOT
+
+#------------------------------------------------------------------------------
+# build lingvo
+#------------------------------------------------------------------------------
+
+FROM wheel-builder as lingvo-builder
+ARG REPO_LINGVO=https://github.com/tensorflow/lingvo.git
+ARG REF_LINGVO=master
+ARG SRC_PATH_LINGVO=/opt/lingvo
+
+COPY --from=tftext-builder /opt/tensorflow-text/tensorflow_text*.whl /opt/
+
+RUN get-source.sh -f ${REPO_LINGVO} -r ${REF_LINGVO} -d ${SRC_PATH_LINGVO}
+
+# build lingvo
+RUN <<"EOT" bash -exu
+set -o pipefail
+
+pushd ${SRC_PATH_LINGVO}
+git fetch origin pull/329/head:pr329
+git cherry-pick --allow-empty pr329
+
+# Disable 2 flaky tests here
+patch -p1 <<"EOF"
+diff --git a/pip_package/build.sh b/pip_package/build.sh
+index ef62c432e..659e78956 100755
+--- a/pip_package/build.sh
++++ b/pip_package/build.sh
+@@ -89,7 +89,7 @@ bazel clean
+ bazel build $@ ...
+ if ! [[ $SKIP_TESTS ]]; then
+   # Just test the core for the purposes of the pip package.
+-  bazel test $@ lingvo/core/...
++  bazel test $@ lingvo/core/... --  -//lingvo/tasks/mt:model_test -//lingvo/core:saver_test
+ fi
+
+ DST_DIR="/tmp/lingvo/dist"
+EOF
+
+pip install tensorflow_datasets==4.9.2 auditwheel tensorflow==2.13.0 /opt/tensorflow_text*.whl
+sed -i 's/tensorflow=/#tensorflow=/'  docker/dev.requirements.txt
+sed -i 's/tensorflow-text=/#tensorflow-text=/'  docker/dev.requirements.txt
+sed -i 's/dataclasses=/#dataclasses=/'  docker/dev.requirements.txt
```

Comment:
The previous stages seem to relate to TF, what is the significance of removing dataclasses as a dep?

Reply:
This is inherited from https://github.com/NVIDIA/JAX-Toolbox/blob/4e4e137afd884ebfa748cec636ea751dc8d7ff7a/.github/container/install_lingvo_aarch64.sh#L41. @joker-eph Do you still remember the reason for doing so?

Reply:
It was needed at some point to install on GH. Maybe that changed, the only way to know is to test it.